rpm/packetfence.spec

```diff
@@ -1,87 +1,43 @@
-# PacketFence RPM SPEC
-#
-# NEW (since git migration):
-#
-#   Expecting a standard tarball with packetfence-<version>/...
-# 
-# BUILDING FOR RELEASE
-# 
-# - Build
-#  - define ver <version>
-#  - define dist based on target distro (for centos/rhel => .el5)
-#  - define rev based on package revision (must be > 0 for proprer upgrade from snapshots)
-# ex:
-# cd /usr/src/redhat/
-# rpmbuild -ba --define 'version 3.3.0' --define 'dist .el5' --define 'rev 1' SPECS/packetfence.spec
-#
-#
-# BUILDING FOR A SNAPSHOT (PRE-RELEASE)
-#
-# - Build
-#  - define ver <version>
-#  - define snapshot 1
-#  - define dist based on target distro (for centos/rhel => .el5)
-#  - define rev to 0.<date> this way one can upgrade from snapshot to release
-# ex:
-# cd /usr/src/redhat/
-# rpmbuild -ba --define 'version 3.3.0' --define 'snapshot 1' --define 'dist .el5' --define 'rev 0.20100506' SPECS/packetfence.spec
-#
-Summary: PacketFence network registration / worm mitigation system
-%global real_name packetfence
-%global perl_version 5.10.1
-Name: %{real_name}-source
-Version: %{ver}
-Release: %{rev}%{?dist}
-License: GPL
-Group: System Environment/Daemons
-URL: http://www.packetfence.org
-BuildRoot: %{_tmppath}/%{real_name}-%{version}-%{rev}-root
-# disables the creation of the debug package for our setuid C wrapper
-%define debug_package %{nil}
-
-Packager: Inverse inc. <support@inverse.ca>
-Vendor: PacketFence, http://www.packetfence.org
-
-# if --define 'snapshot 1' not written when calling rpmbuild then we assume it is to package a release
-%define is_release %{?snapshot:0}%{!?snapshot:1}
-%if %{is_release}
-# used for official releases
-Source: http://www.packetfence.org/downloads/PacketFence/src/%{real_name}-%{version}.tar.gz
-%else
-# used for snapshot releases
-Source: http://www.packetfence.org/downloads/PacketFence/src/%{real_name}-%{version}-%{rev}.tar.gz
-%endif
-
-# Log related globals
-%global logfiles packetfence.log snmptrapd.log pfdetect pfmon security_event.log httpd.admin.audit.log
-%global logdir /usr/local/pf/logs
+#==============================================================================
+# Variables
+#==============================================================================
+# use 'global' variables (vs 'define' with local scope)
+%global     builddoc 0
+%global     perl_version 5.10.1
+%global     logfiles packetfence.log snmptrapd.log pfdetect pfmon security_event.log httpd.admin.audit.log
+%global     logdir /usr/local/pf/logs
+%global     debug_package %{nil}
+
+
+#==============================================================================
+# Main package
+#==============================================================================
+Name:       packetfence
+Version:    9.1.9
```

Comment:
Can it be a variable ?

Reply:
Yes. From my point of view, we should define upstream version in one location (like `conf/pf-release`) and use a script to upgrade all other locations like:
- rpm/packetfence.spec
- rpm/source
- rpm/packetfence-release/packetfence-release.spec
- debian/changelog